cpp/tests/io/csv_test.cu

```diff
@@ -110,6 +221,36 @@ void expect_column_data_equal(std::vector<T> const& lhs, cudf::column_view const
   EXPECT_THAT(cudf::test::to_host<T>(rhs).first, ::testing::ElementsAreArray(lhs));
 }
 
+void write_csv_helper(std::string const& filename,
+                      cudf::table_view const& table,
+                      bool include_header,
+                      std::vector<std::string> const& names = {})
+{
+  // NOTE sink_info, na, and metadata lifetimes must extend past write_csv call since
+  // write_csv_args is non-owning
+  cudf_io::sink_info const sink{filename};
+  std::string const na{"null"};  // why doesn't this have a default?
+  cudf_io::table_metadata metadata{};
+
+  if (not names.empty()) {
+    metadata.column_names = names;
+  } else {
+    // generate some dummy column names
+    int i                  = 0;
+    auto const num_columns = table.num_columns();
+    metadata.column_names.reserve(num_columns);
+    std::generate_n(std::back_inserter(metadata.column_names), num_columns, [&i]() {
+      return std::string("col") + std::to_string(i++);
+    });
+  }
+
+  int const rows_per_chunk{1};  // what should this be set to? how would I know?
```

Comment:
Still have the same question about `rows_per_chunk` here

Reply:
As already mentioned, per legacy code logic and requirements `rows_per_chunk` gets adjusted to nearest non-zero multiple of `8`. Added that comment in the code.